### DISCUSSION TITLE
SCIM support

### DISCUSSION THREAD

**Discussion by schewara** [dani-garcia/vaultwarden#3842] [Ideas]
Currently it is rather painful to set up users and support for SCIM would be a good option to get the users provisioned

Bitwarden itself does already support it and would be a really awesome to be also supported by Vaultwarden

https://bitwarden.com/help/about-scim/

**Comment by BlackDex** [comment]
This is an Enterprise/Bitwarden Licensed feature, so, this will not going to be part of Vaultwarden any time soon.

**Reply by schewara** [reply]
@BlackDex Thanks for the quick reply and the clarifications.

From what I saw, It seems that https://github.com/dani-garcia/vaultwarden/pull/3154 might also be helpful for getting users onboarded easier.

To bring a bit more context to it, our main problem with setting up is the following unintuitive workflow:

- a user gets an invite to an Organisation which requires 2FA
- the user clicks on the invite and creates his user and needs to set up 2FA
  (as there is no global 2FA enforcing policy, this step gets often missed, as it is not required during the account creation steps)
- once the user has his account set up, he has to go back to the mail and again click the invite, which requires another login to join the organization.

This causes a lot of confusion when using vaultwarden for the first time.

Maybe I missed something which could improve the initial experience, so if there are some pointers I am happy to learn about them.

**Reply by BlackDex** [reply]
That seems to be correct. Since Bitwarden does not provide a way to configure MFA/2FA during the account creation this is just the flow on how Bitwarden works. There is nothing we can do about this from the server side.

**Comment by martinmolt** [comment]
Any update on this, as SSO is now available (and working 🥳) in :testing? 
It's a bit frustrating, that you have to invite users into organizations even when they signed in via SSO (as described in #6192 )

Thanks! :)
